examples/mnist/train.cpp

```diff
@@ -79,20 +87,20 @@ void train_lenet(std::string data_dir_path) {
 
     std::cout << "start training" << std::endl;
 
-    progress_display disp(train_images.size());
+    progress_display disp((unsigned long)train_images.size());
     timer t;
```

Comment:
I wished to do so but in the file `tiny_cnn/io/display.h`, the prototype of `progress_display` class constructor is this.

```
  explicit progress_display( unsigned long expected_count_,
                             std::ostream & os = std::cout,
                             const std::string & s1 = "\n", //leading strings
                             const std::string & s2 = "",
                             const std::string & s3 = "" )
```

So that I used `unsigned long` cast code to remove a warning.